contracts/voting-verifier/src/events.rs

```diff
@@ -137,6 +138,12 @@ fn parse_message_id(
 
             Ok((id.tx_hash_as_hex(), id.event_index))
         }
+        MessageIdFormat::Base58SolanaTxDigestAndEventIndex => {
+            let id = Base58SolanaTxDigestAndEventIndex::from_str(&message_id)
+                .map_err(|_| ContractError::InvalidMessageID(message_id.into()))?;
```

Comment:
I wonder about providing more error info. But maybe it doesn't worth it, as we wouldn't get much more information anyways.